cg/services/illumina_services/illumina_post_processing_service/illumina_post_processing_service.py

```diff
@@ -27,36 +30,51 @@ def __init__(self, status_db: Store, housekeeper_api: HousekeeperAPI, dry_run: b
 
     @staticmethod
     def store_illumina_flow_cell(
-        flow_cell: FlowCellDirectoryData,
+        flow_cell_dir_data: FlowCellDirectoryData,
         store: Store,
     ) -> IlluminaFlowCell:
         """
         Create flow cell from the parsed and validated flow cell data.
         And add the samples on the flow cell to the model.
         """
-        model: str | None = flow_cell.run_parameters.get_flow_cell_model()
+        model: str | None = flow_cell_dir_data.run_parameters.get_flow_cell_model()
         new_flow_cell = IlluminaFlowCell(
-            internal_id=flow_cell.id, type=DeviceType.ILLUMINA, model=model
+            internal_id=flow_cell_dir_data.id, type=DeviceType.ILLUMINA, model=model
         )
         return store.add_illumina_flow_cell(new_flow_cell)
 
     @staticmethod
-    def store_illumina_sequencing_metrics(flow_cell: IlluminaFlowCell) -> None:
+    def get_illumina_sequencing_run(
+        flow_cell_dir_data: FlowCellDirectoryData,
+    ) -> IlluminaSequencingRun:
         """Store illumina run metrics in the status database."""
-        pass
+        metrics_service = IlluminaMetricsService()
+        return metrics_service.create_illumina_sequencing_run(flow_cell_dir_data)
 
-    @staticmethod
-    def store_illumina_sample_sequencing_metrics():
+    def get_illumina_sample_sequencing_metrics(
+        self,
+        flow_cell_dir_data: FlowCellDirectoryData,
+    ) -> list[IlluminaSampleSequencingMetrics]:
         """Store illumina sample sequencing metrics in the status database."""
-        pass
+        metrics_service = IlluminaMetricsService()
+        return metrics_service.create_sample_sequencing_metrics_for_flow_cell(
+            flow_cell_directory=flow_cell_dir_data.path, store=self.status_db
+        )
 
-    def store_illumina_flow_cell_data(self, flow_cell: FlowCellDirectoryData) -> None:
+    def store_illumina_flow_cell_data(self, flow_cell_dir_data: FlowCellDirectoryData) -> None:
         """Store flow cell data in the status database."""
         flow_cell: IlluminaFlowCell = self.store_illumina_flow_cell(
-            flow_cell=flow_cell, store=self.status_db
+            flow_cell_dir_data=flow_cell_dir_data, store=self.status_db
+        )
+        sequencing_run: IlluminaSequencingRun = self.get_illumina_sequencing_run(flow_cell_dir_data)
+        sample_metrics: list[IlluminaSampleSequencingMetrics] = (
+            self.get_illumina_sample_sequencing_metrics(flow_cell_dir_data)
         )
-        self.store_illumina_sequencing_metrics(flow_cell)
-        self.store_illumina_sample_sequencing_metrics()
+        sequencing_run.device = flow_cell
+        sequencing_run.sample_metrics = sample_metrics
+        self.status_db.add_illumina_flow_cell(flow_cell)
+        self.status_db.add_illumina_sequencing_run(sequencing_run)
+        self.status_db.add_illumina_sample_metrics(sample_metrics)
```

Comment:
This is starting to look a little bit messy to me. Can we standardise the naming of the functions that create database models? And can we also move the adding to the database to be within those functions?

Ideally I would like store_illumina_flow_cell_data to just call three methods (named simliarly) and then commit to the database. I think that would improve the readability

Reply:
outdated, see new implementation.